src/kdbindings/connection_evaluator.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <functional>
+#include <list>
+#include <mutex>
+
+namespace KDBindings {
+
+/**
+ * @brief Manages and evaluates deferred Signal connections.
+ *
+ * The ConnectionEvaluator class is responsible for managing and evaluating connections
+ * to Signals. It provides mechanisms to delay and control the evaluation of connections.
+ * It therefore allows controlling when and on which thread slots connected to a Signal are executed.
+ *
+ * @see Signal::connectDeferred()
+ */
+class ConnectionEvaluator
+{
+
+public:
+    ConnectionEvaluator() = default;
+
+    // ConnectionEvaluators are not copyable, as it is designed to manage connections,
+    // and copying it could lead to unexpected behavior, including duplication of connections and issues
+    // related to connection lifetimes. Therefore, it is intentionally made non-copyable.
+    ConnectionEvaluator(const ConnectionEvaluator &) noexcept = delete;
+
+    ConnectionEvaluator &operator=(const ConnectionEvaluator &) noexcept = delete;
+
+    // ConnectionEvaluators are not moveable, as they are captures by-reference
+    // by the Signal, so moving them would lead to a dangling reference.
+    ConnectionEvaluator(ConnectionEvaluator &&other) noexcept = delete;
+
+    ConnectionEvaluator &operator=(ConnectionEvaluator &&other) noexcept = delete;
+
+    /**
+     * @brief Evaluate and execute deferred connections.
```

Comment:
I'd argue that no evaluation happens here. Any invocation of a slot that got enqueued in the list will be unconditionally executed.
`dispatchPendingSlots()` maybe?

Reply:
Hm, interesting point.
The name `ConnectionEvaluator` is meant to mimic the naming of the `BindingEvaluator`.
It's a fair point that the Connection is less "evaluated" and more "executed". However, I think that it's still fair to call it `evaluate`, as it's the expected API when compared to the BindingEvaluator. Also, you're still calling a function (i.e. slot) and I think "evaluating" a function should be understandable.

It doesn't make any sense to say "evaluate **and** execute". We should therefore probably document this as "evaluate deferred connections" **or** "execute deferred connections". I'd be fine with either.

Reply:
> and I think "evaluating" a function should be understandable.

I buy that reasoning 👍

But yeah, the doc shouldn't talk about both executing *and* evaluating. Either or.